### PR TITLE
Add tests and fix bugs

### LINKED ISSUE #249: [PE-D][Tester B] Incorrect UML Notation in Developer Guide Diagrams
**Description**:  
Two UML diagrams in the Developer Guide contain incorrect notations. The `Logic` component diagram and the `Parser` diagram use double-headed arrows, which is not a valid UML notation. Instead, an association label should be used to indicate the direction and nature of the association between components.

**Steps to Reproduce**:
1. Open the Developer Guide.
2. Navigate to the `Logic component` section and locate the first and third diagrams.

**Expected Result**:  
All UML diagrams should follow correct UML notation. In this case, valid association lines with labels should be used to represent relationships instead of using double-headed arrows.

**Actual Result**:  
Double-headed arrows are used to indicate relationships, which is incorrect UML notation.

**Proposed Solution**:
- Update the UML diagrams to use proper association labels instead of double-headed arrows to ensure compliance with UML standards.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: TechRaj/ped#16

### LINKED ISSUE #269: Add more tests
### **Description**
Test for potential bugs especially on parameters

### LINKED ISSUE #271: Filter command gives incorrect error message for range
### **Description**
Filter command gives incorrect error message.

### **Steps to Reproduce**
Examples:
`filter b/-3--1`
`filter b/-4-5`

### **Expected Result**
Error messages should state that the age given is invalid.

### **Actual Result**
The range is valid but the error message says that the range is not. 
image
image

### **Suggested Improvement**
Widen the scope of error message. (Similar to #214 )

### LINKED ISSUE #272: Filter command gives incorrect error message for negative integer
### **Description**
Filter command gives incorrect error message.

### **Steps to Reproduce**
Examples:
`filter b/-3`

### **Expected Result**
Error messages should state that the age given is invalid and/or range is invalid.

### **Actual Result**
Error message shows that the age should be an integer but the age is indeed an integer.
image

### **Suggested Improvement**
Widen the scope of error message. (Similar to #214 )

### PR DESCRIPTION
**Pull request by Bonzzz3**
**Key Changes**
- Add assertions and more tests (Closes #269 )
- Fix DG "Implementation" still showing in table of contents (#266 )
- Widen the error message for filter age (Fixes #271 , Fixes #272 )
- Fix incorrect UML notation (Fixes #249 )
- Update AboutUs page

Code cov less because of assertions added, already add test for it though

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java [hunk 1]
```diff
@@ -25,6 +25,8 @@ public class FilterCommandParser implements Parser<FilterCommand> {
     public static final String INCORRECT_AGE = "Age should be an integer";
     public static final String INCORRECT_DATE_FORMAT = "Appointment date should be form dd/MM/yyyy";
     public static final String INCORRECT_RANGE = "Please provide range in format: Lower bound - Upper bound";
```
Comment (myrilla12) [inline src/main/java/seedu/address/logic/parser/FilterCommandParser.java:27]: Is it necessary to keep INCORRECT_RANGE?

**Review by myrilla12 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/FilterCommandParser.java (hunk 1 above)
Comment (Bonzzz3) [reply src/main/java/seedu/address/logic/parser/FilterCommandParser.java:27]: Yes it's used for appointment range also. I can remove INCORRECT_AGE tho

**Review by ritvi12 — APPROVED** [review]
LGTM!

**Pull request merged**